optuna/pruners/percentile.py

```diff
@@ -75,11 +75,11 @@ def __init__(self, percentile, n_startup_trials=5, n_warmup_steps=0):
         self.n_startup_trials = n_startup_trials
         self.n_warmup_steps = n_warmup_steps
 
-    def prune(self, storage, study_id, trial_id, step):
-        # type: (BaseStorage, int, int, int) -> bool
+    def prune(self, study, trial, step):
+        # type: (Study, structs.FrozenTrial, int) -> bool
         """Please consult the documentation for :func:`BasePruner.prune`."""
 
-        n_trials = storage.get_n_trials(study_id, structs.TrialState.COMPLETE)
+        n_trials = study.get_n_trials(state=structs.TrialState.COMPLETE)
```

Comment:
Although we can also use `trials` property method, but the cost to call `storage.n_trials` is basically cheaper in RDB storage. So I defined `get_n_trials` in Study.

Reply:
IMO, `study.get_n_trials` method can be removed.
This method is only beneficial only when the number of completed trials is less than `n_warmup_steps`.
However, in such case, the cost of `study.trials` tends to relatively small, so I prefer interface simplicity (i.e., doesn't add the new method to `Study`) to the performance merit.

Reply:
You are right. I'll remove `get_n_trials`.

Reply:
Thank you!